eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -185,7 +185,7 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, wallet: EclairWa
         Logs.withMdc(diagLog)(Logs.mdc(category_opt = Some(Logs.LogCategory.CONNECTION))) {
           log.info("connection lost")
         }
-        if (d.channels.isEmpty) {
+        if (!nodeParams.forceReconnect(remoteNodeId) && d.channels.isEmpty) {
```

Comment:
I think you also need to update this condition: https://github.com/ACINQ/eclair/blob/a8d4e07bdde9b5f28ebad291ac3dbef9cc97c759/eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala#L200 to keep the connection open even if there are no channels left?

Reply:
To think about it, `Peer` logic seems rather loaded as is, OTOH it occured to me that any plugin can easily request reconnection by itself on getting `PeerDisconnected` (for example by obtaining `Router` data and extracting address from there, or by maintaining its own db).

Maybe it makes sense to get rid of a whole `ConnectionControlPlugin` logic?

Reply:
If we can get rid of it without impacting plugins too much, then hell yes, less complexity to deal with inside the peer ;)

Reply:
For now the only plugin (that I'm aware of) using this is mine, and even that one is not released yet, so it's a good time to roll this feature back. I'll proceed with PR if that's OK.

Reply:
Perfect, that sounds good to me!